rust/scx_utils/src/load_balance.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+//! # SCX Load Balancer
+//!
+//! A crate providing abstractions for implementing load balancing logic
+//! between scheduling domains.
+//!
+//! Initializing LoadBalancer
+//! -------------------------
+//!
+//! LoadBalancer objects are created from LoadAggregator objects. As the name
+//! suggests, LoadAggregator objects aggregate load passed by the caller, and
+//! create a LoadBalancer object that contains load and weights for each
+//! domain; adjusted for any infeasible weights.
+//!
+//! Note that the abstractions here are meant to be generic across schedulers.
+//! LoadAggregator assumes nothing about the scheduling domains being passed to
+//! it. For example, they may span layers defined in a scheduler, domains
+//! specified by the user via cpumask strings, or domains that span L3 caches.
+//! The crate simply aggregates load according to domain with IDs specified by
+//! the caller, computes global load sums, and adjusts any infeasible weights
+//! such that all infeasible scheduling entities still get their full duty
+//! cycle.
+//!
+//! Assume we're on a 16-core (32-CPU) host with two core complexes:
+//!
+//!```
+//!     let mut aggregator = LoadBalancer::aggregator(32, false);
```

Comment:
Is this a common pattern? Wouldn't `LoadAggregator::new(32, false)` be more usual? Is it to lower the number of structs that users have to import?

Reply:
My thinking was more so that `LoadAggregator` is only really useful in the context of creating a `LoadBalancer`. In my experience it's fairly normal to do this for builder patterns, but I'm fine with changing it if that's your preference.